### DISCUSSION TITLE
2021-12-06 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#866] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Mathesar release plan update](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748045)
- [Frontend planning](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748047)
- [Backend planning](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748048)
- [Design planning](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748052)
- [Product planning](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748053)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1748054)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1755855)
- [Dominykas' check-in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1756526)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1757661)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1760555)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1760154)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/866#discussioncomment-1759392)

## Weekly check-in
@dmos62 @ghislaineguerin @mathemancer @pavish @seancolsen @silentninja please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next week? Do you anticipate any blockers?
- I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Mathesar release plan update

The core team met earlier this week for a "State of Mathesar meeting" to check in on Mathesar's progress. Outcomes of the meeting include:
- We've cut many more milestones from our alpha release roadmap. We now plan to release after we are done with basic functionality for tables, views, and data types. Our goal is for early users to be able to tinker with the project and offer feedback. Please see our [GitHub milestone roadmap](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open) for the latest plan.
- We're considering having monthly meetings to check in on progress since many of us expressed finding the meeting useful.

Full meeting notes can be found on [our wiki](https://wiki.mathesar.org/meeting-notes/2021-11), as usual.

**Comment by kgodey** [comment]
## Frontend planning

Since @pavish is probably unavailable this week, it's going to be a light week for the frontend. Our priorities should continue to be making progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone.

@seancolsen, you have a number of PRs in review. Would it speed up your work for me or someone else on the team to get some PRs reviewed and merged, or are you okay waiting for @pavish to be available?

**Reply by seancolsen** [reply]
@kgodey 

- **Open PRs**:
    - Yeah it would be nice if you or someone else could help get some of mine merged.
    - #836 would be the first one to merge.
    - Then either #857 or #852.
    - The other PRs can wait longer since I'm unlikely to want to stack any work on top of them.

- **Priorities**:
    - I'm currently working on #445. After I'm done with that, I'm not sure what else I would take within [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8). Potentially #497, but it's assigned to @pavish.
    - I'd be inclined to work on:
        - #826
        - #742
        - #475

**Reply by kgodey** [reply]
I've assigned those PRs to myself and I'll reassign them in case I get any other volunteer reviewers.

I think it's fine if you reassign #497 to yourself. The rest of the issues you mentioned sound good too.

**Comment by kgodey** [comment]
## Backend planning

We're continuing to work on the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone and have made significant progress towards working out extendable APIs for filtering and grouping as well as support for two additional data types.

### Priorities

This week, our first priority should be to get following major PRs merged. Please prioritize reviewing these and making follow up changes.
- #628
- #822
- #837
- #862
- #865

Our next priority should be to finalize and implement the new API error structure: The spec is currently being tracked in #560. Once it is finalized, we can open a new issue for implementation. This is not directly related to data types, but it's a fundamental structural issue that we should handle sooner rather than later.

Our third priority is to continue making progress on data types - supporting additional data types, as well as supporting filtering, grouping, and display options for all of them.

### Questions
These questions are for @dmos62 @mathemancer @silentninja.

- Do the priorities for this week sound good to you?
- Any concerns or blockers?
- Any pending discussions we need to resolve or feedback you need from anyone else?

**Reply by silentninja** [reply]
>  Do the priorities for this week sound good to you?

Yes, seems right.

>  Any concerns or blockers?

Nothing at the moment

> * Any pending discussions we need to resolve or feedback you need from anyone else?

None

**Reply by dmos62** [reply]
>  Do the priorities for this week sound good to you?

:+1:

>  Any concerns or blockers?

Nothing at the moment.

> Any pending discussions we need to resolve or feedback you need from anyone else?

@mathemancer @silentninja The discussion on how queries that require both grouping and filtering (like `duplicates_only`, but more general) could be constructed would benefit from more brainstorming: https://github.com/centerofci/mathesar/discussions/853.

**Reply by mathemancer** [reply]
> Do the priorities for this week sound good to you?

I think they make sense.

> Any concerns or blockers?

There are some dependencies we're all aware of there.

> Any pending discussions we need to resolve or feedback you need from anyone else?

I'd like to make sure the grouping API changes make sense, and get the front end synced up with them.  Any grouping methods or changes we want to add should be easier if we're happy with that different paradigm.

I'd also like general feedback on the concept introduced by that PR (#862 ) of using CTEs as our main composable building block for piecing together SQL logic.  I think we're going to need to be really conscious of how we do that, since it's difficult to make reusable SQL code pieces.

**Reply by kgodey** [reply]
Thanks everyone.

**Comment by kgodey** [comment]
## Design planning

Our priorities for design this week are:
- Get handling `NULL` values reviewed and finalized: https://github.com/centerofci/mathesar-wiki/pull/13
- Get design for Foreign Keys reviewed and finalized: https://github.com/centerofci/mathesar-wiki/pull/12
- Make progress on design for:
  - https://github.com/centerofci/mathesar/issues/455
  - https://github.com/centerofci/mathesar/issues/466

Everyone, please prioritize re-reviewing the two design PRs above.

@ghislaineguerin does this plan sound good to you?

**Reply by ppii775** [reply]
@kgodey @ghislaineguerin 
An update on my end. I'm still working on #466.
Last week and the upcoming week are final weeks, but I will have more free time after Thursday 12/9.

**Reply by ghislaineguerin** [reply]
@kgodey This plan sounds good.

**Reply by kgodey** [reply]
@ppii775 Thanks for the update and good luck with your finals!

**Comment by kgodey** [comment]
## Product planning

My main aims this week are to:
- Define Mathesar Types better: https://github.com/centerofci/mathesar-wiki/issues/14
- Define data sources in Views better and update other Views tickets: #806
- Create backend and frontend issues for merged design specs if time permits

I expect to have a light week as far as product work goes since I'm focused on the CCI board presentation.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2021-12-06 to 2021-12-10 | Mukesh |
| 2021-12-13 to 2021-12-17 | Sean |
| 2021-12-20 to 2021-12-23 | Dominykas |
| 2021-12-24 to 2022-01-02 | *CCI closed; team is unavailable.* |
| 2022-01-03 to 2022-01-07 | Ghislaine |
| 2022-01-10 to 2022-01-14 | Brent |
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by silentninja** [comment]
> How did the last week go? What went well, what could have gone better?

Was okay, Recovered fully from illness and was finally able to complete the timestamp datatype issue and work on few pending issues

> What are you planning to work on over the next week? Do you anticipate any blockers?

Aside from the priorities listed above, I am planning to work on 
- #430
- #428 and #433
- #839
I might be prioritizing working on filtering option issues rather than #839 based on how the issue progresses. 

> I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these > sections helpful for you, and is the format helpful?

Yes, definitely prefer this.

> Anything else you'd like to add?

I happened to talk with a few journalism students who would find Mathesar useful for their data journalism class. I will be getting feedback before the holidays on how they plan to use it, features they might need for their data journalism class. I am also planning to write a wiki page that gives an overview of each module, so that it is easier to find the necessary details about how it relates to the project and where to find/write the tests for each

**Reply by ghislaineguerin** [reply]
@silentninja let me know if you need help w/ setting up a  user research session w the students. I'd love to help.

**Reply by silentninja** [reply]
Sure! They have exams going on right now, I will see if I can set a session once the holidays are over.

**Comment by dmos62** [comment]
> How did the last week go? What went well, what could have gone better?

Finished up and submitted the filtering PR. Spent a long time in discussions and thinking about filtering/grouping architecture, since we seem to have arrived at a place where we can't extend this setup without major refactors.

Got some feedback about my raised discussions being not immediately actionable or too abstract. Will work on that.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Kriti set this week's priorities to PR review and API error spec. When done with that will get back to filtering: the `duplicates_only` query needs to be supported. Expecting some filtering and grouping related discussions too.

> I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?

Yes, the sections/format are helpful.

> Anything else you'd like to add?

No.

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- Going well.
- I found the State of Mathesar meeting particularly useful, and I look forward to the next one.

What are you planning to work on over the next week? Do you anticipate any blockers?

- Not exactly sure. Will discuss further in the front end thread.

I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?

- This seems like an improvement so far.

Anything else you'd like to add?

- Nope.

**Comment by mathemancer** [comment]
> How did the last week go? What went well, what could have gone better?

Things went pretty well.  I think we got some clarity on various design issues, and @dmos62 started some much-needed discussions, especially about filtering.  I was also happy to finally get a PR in for grouping API changes, and happy with the refactoring.  If we can keep moving in the direction of focusing on reusable SQL logic, we'll be much more productive in the long run (I hope).

> What are you planning to work on over the next week? Do you anticipate any blockers?

My top priority is reviewing PRs and design updates.  Given validation on the approach in #862 , I'll keep working on grouping to finish off numeric grouping as well as ranged grouping for any orderable types.  I suspect I'll also be unable to resist participating in https://github.com/centerofci/mathesar-wiki/issues/14 , or at least discussions around that.

> I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?

I found those sections quite helpful, and the follow up discussions seemed well-focused.

> Anything else you'd like to add?

Not at the moment.

**Comment by ghislaineguerin** [comment]
> How did the last week go? What went well, what could have gone better?

I had hoped to close the foreign key spec PR, but I had additional questions about settings and that slowed me down. I also misunderstood a requirement for settings override which ended up costing me more time. However, it is also good that I was able to start work on views. 

> What are you planning to work on over the next week? Do you anticipate any blockers?

Listing of views. No blockers that I can think of.

> I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?

It is helpful.

> Anything else you'd like to add?

Not at the moment.

**Comment by kgodey** [comment]
> How did the last week go? What went well, what could have gone better?

I was out for a lot of it, not much to report. I'm glad we have a new plan for the alpha release.

> What are you planning to work on over the next week? Do you anticipate any blockers?

- Mathesar board presentation
- PR review for backend and frontend
- Define Mathesar Types better: https://github.com/centerofci/mathesar-wiki/issues/14
- Define data sources in Views better and update other Views tickets: #806
- Create backend and frontend issues for merged design specs if time permits

> I'm trying out a new format where each area (frontend, backend, design, product) gets a separate planning post. Are these sections helpful for you, and is the format helpful?

I think it's easier to read. It also seems to be encouraging more discussion, which was my hope.
